eldar/entry.py

```diff
@@ -34,7 +34,7 @@ def evaluate(self, doc):
             else:
                 res = False
         else:
-            res = True if re.search(fr"\b{self.query}\b", doc) else False
+            res = True if re.search(fr"\\b?[:-_.]?{self.query}[:-_.]?\\b?", doc) else False
```

Comment:
So the issue here is that since both the `\b` and `[:-_.]` are optional (`?`), we could end up just matching on a substring (e.g. if the query was `"fuse"` it would match things like `"confuse"`). To get around this I think that we're gonna need to make a function called like `generate_search_regex`. 

This will then just be something like:

```python
import string

def generate_search_regex(query: str) -> str:
    """
    some description of the function
    """
    search_query = f"{query}"
    if search_query[0] not in string.punctuation:
        search_query = f"\b{search_query}"

    if search_query[-1] not in string.punctuation:
        search_query = f"{search_query}\b"

    return search_query   
```